src/avatar.cpp

```diff
@@ -237,8 +237,15 @@ void avatar::memorial( std::ostream &memorial_file, const std::string &epitaph )
     memorial_file << dump_memorial();
 }
 
-    // Ignore list that lasts for the session (rather than the single activity)
-    void player_activity::ignore_distraction_this_session( distraction_type type )
+// Ignore list that lasts for the session (rather than the single activity)
+bool avatar::is_distraction_ignored_this_session( distraction_type type ) const
+{
+    ignored_distractions_this_session( rhs.ignored_distractions_this_session );
```

Comment:
I'm not sure to understand what you're trying to do here (and the line below). You already have the comparison at the end of this function and that's all you really need. 

Basically you can copy the implementation around the member field `ignored_distractions` of the `player_activity` class, just making sure it's not saved anywhere. You also have an example in `player_activity::is_distraction_ignored` which works the same as what you are trying to achieve in this function.